src/db_ops/spark_db_utils.py

```diff
@@ -8,17 +8,43 @@
 
 def create_namespace_if_not_exists(
         spark: SparkSession,
-        namespace: str = "default"
+        namespace: str = "default",
+        append_target: bool = False,
 ) -> None:
 
     """
     Create a namespace in the Spark catalog if it does not exist.
+    
+    If append_target is True, automatically prepends the target name (user/tenant)
+    from the Spark warehouse configuration to create {target_name}_{namespace}.
 
     :param spark: The Spark session.
-    :param namespace: The name of the namespace. Default is "default".
+    :param namespace: The name of the namespace.
+    :param append_target: If True, prepends target name from warehouse directory
+                         (e.g., "john_default" or "research_team_experiments").
+                         If False, uses namespace as-is.
     :return: None
     """
-
+    try:
+        # Extract user/tenant name from warehouse directory if append_target is enabled
+        if append_target:
+            warehouse_dir = spark.conf.get('spark.sql.warehouse.dir', '')
```

Comment:
When would this not be set?

Reply:
Users technically can create their own Spark session, but this is highly unlikely. It’s here as a safeguard.

Reply:
:+1: